libs/gi/db/src/Database/DataManagers/OptConfigDataManager.ts

```diff
@@ -307,3 +308,33 @@ export function handleArtSetExclusion(
   }
   return artSetExclusion
 }
+
+function mapFilterInplace<T>(xs: T[], f: (x: T) => T | undefined): T[] {
```

Comment:
Can you add some comments to what the assumption of this function is, and why we should use it over the regular `map.filter`? would also be nice to annotate the variables as well

Reply:
We can also consider moving this function to comon/util function, if we can see more places where we can use this utility